src/Altinn.App.Api/Controllers/InstancesController.cs

```diff
@@ -683,6 +691,10 @@ [FromRoute] Guid instanceGuid
         ProcessChangeResult startResult = await _processEngine.GenerateProcessStartEvents(processStartRequest);
 
         targetInstance = await _instanceClient.CreateInstance(org, app, targetInstance);
+        foreach (var instanceEvent in startResult.ProcessStateChange?.Events ?? [])
```

Comment:
These loops are needed because `GenerateProcessStartEvents` is creating the events, at which point `InstanceId` is null..

Reply:
Was this a bug before this fix?

Reply:
I don't think so, as it started to happen after switching to the new endpoint.. Don't know why, will do some digging to make sure